segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/SystemJournal.java

```diff
@@ -263,60 +375,107 @@ public void commitRecords(Collection<SystemJournalRecord> records) throws ChunkS
-                    // Read contents.
-                    byte[] contents = getContents(snapshotFile);
-                    systemSnapshot = SYSTEM_SNAPSHOT_SERIALIZER.deserialize(contents);
-                } catch (EOFException e) {
-                    log.warn("SystemJournal[{}] Incomplete snapshot found, skipping {}.", containerId, snapshotFile);
-                }
-                if (null != systemSnapshot) {
-                    log.debug("SystemJournal[{}] Processing system log snapshot {}.", containerId, systemSnapshot);
-                    // Initialize the segments and their chunks.
-                    for (SegmentSnapshotRecord segmentSnapshot : systemSnapshot.segmentSnapshotRecords) {
-                        // Update segment data.
-                        segmentSnapshot.segmentMetadata.setActive(true)
-                                .setOwnershipChanged(true)
-                                .setStorageSystemSegment(true);
-                        segmentSnapshot.segmentMetadata.setOwnerEpoch(epoch);
-
-                        // Add segment data.
-                        txn.create(segmentSnapshot.segmentMetadata);
-
-                        // make sure that the record is marked pinned.
-                        txn.markPinned(segmentSnapshot.segmentMetadata);
-
-                        // Add chunk metadata and keep track of start offsets for each chunk.
-                        long offset = segmentSnapshot.segmentMetadata.getFirstChunkStartOffset();
-                        for (ChunkMetadata metadata : segmentSnapshot.chunkMetadataCollection) {
-                            txn.create(metadata);
-
-                            // make sure that the record is marked pinned.
-                            txn.markPinned(metadata);
-
-                            chunkStartOffsets.put(metadata.getName(), offset);
-                            offset += metadata.getLength();
-                        }
-                        found = true;
-                    }
-                    break;
+            // Read contents.
+            try {
+                byte[] snapshotContents = getContents(snapshotFileName);
+                val systemSnapshot = SYSTEM_SNAPSHOT_SERIALIZER.deserialize(snapshotContents);
+                log.debug("SystemJournal[{}] snapshot found and parsed. {}", containerId, persisted);
+                log.debug("SystemJournal[{}] Done finding snapshots.", containerId);
+                return systemSnapshot;
+            } catch (Exception e) {
+                String object = persisted.toString();
+                val ex = Exceptions.unwrap(e);
+                if (ex instanceof EOFException) {
+                    log.error("SystemJournal[{}] Incomplete {} found, skipping {}.", containerId, "snapshot", object, e);
+                } else if (ex instanceof ChunkNotFoundException) {
+                    log.error("SystemJournal[{}] Missing {}, skipping {}.", containerId, "snapshot", object, e);
```

Comment:
this and the above should be WARN. The difference between ERROR and WARN is that in WARN you may be able to handle it, but in a sub-optimal way. ERROR means you can't do anything about it - that's appropriate below.

Reply:
Fixed.